pre_commit_all_the_way_down.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import re
+from pathlib import Path
+from re import Match
+from sys import exit
+from tempfile import TemporaryDirectory
+from textwrap import dedent, indent
+from typing import Optional, Sequence
+
+import sh
+
+BLOCK_TYPES = "(code|code-block|sourcecode|ipython)"
+PY_LANGS = "(python|py|sage|python3|py3|numpy)"
+DOCTEST_TYPES = "(testsetup|testcleanup|testcode)"
+RST_RE = re.compile(
+    rf"(?P<before>"
+    rf"^(?P<indent> *)\.\. ("
+    rf"jupyter-execute::|"
+    rf"python-script::|"
+    rf"{BLOCK_TYPES}:: {PY_LANGS}|"
+    rf"{DOCTEST_TYPES}::.*"
+    rf")\n"
+    rf"((?P=indent) +:.*\n)*"
+    rf"\n*"
+    rf")"
+    rf"(?P<code>(^((?P=indent) +.*)?\n)+)",
+    re.MULTILINE,
+)
+INDENT_RE = re.compile("^ +(?=[^ ])", re.MULTILINE)
+TRAILING_NL_RE = re.compile(r"\n+\Z", re.MULTILINE)
+
+
+pre_commit = sh.Command("pre-commit").bake("run", "--files")
+
+
+def fix_block(block: str, filename: str):
+    """
+    Fix a code block.
+
+    Parameters
+    ----------
+    block : str
+        The block to fix
+    filename : str
+        The name of the file it originates from (used for logging)
+    lineno : int
+        The linenumber in `filename` where the block starts.
+
+    Notes
+    -----
+    The code block should have no leading indentation and be valid
+    Python.
+    """
+    with TemporaryDirectory(dir=".") as d:
```

Comment:
Any particular reason you need to specify `dir="."`? just curious

Reply:
I have changed it; I thought pre-commit could only run in local folders but I was wrong